pynetbox/core/query.py

```diff
@@ -262,6 +270,24 @@ def _make_call(
         else:
             raise RequestError(req)
 
+    def concurrent_get(self, ret, page_size, page_offsets):
+        futures_to_results = []
+        with cf.ThreadPoolExecutor(max_workers=4) as pool:
+            for offset in page_offsets:
+                new_params = {"offset": offset, "limit": page_size}
+                futures_to_results.append(
+                    pool.submit(self._make_call, add_params=new_params)
+                )
+
+            for future in cf.as_completed(futures_to_results):
+                try:
+                    result = future.result()
+                except Exception:
+                    # TODO: Raise exception with preferred method
```

Comment:
I'd like to hear your thoughts on wait to do here @zachmoody - raise a `RequestError` and blow up the whole thing? Or perhaps a error log.

Reply:
Admittedly haven't done a lot with thread, but what happens without the try-block?

Reply:
Without the try block if the individual thread (one page of the API call) encountered an exception, that exception will be raised for the whole job.

Reply:
Yeah, I think that's ok when the next best alternative (imo) is just catching and re-raising here. Kinda feel like getting an incomplete return should be terminal.

Reply:
Done